packages/ozone/src/verification/service.ts

```diff
@@ -0,0 +1,192 @@
+import { Selectable } from 'kysely'
+import {
+  $Typed,
+  AppBskyActorDefs,
+  AtUri,
+  ToolsOzoneModerationDefs,
+  ToolsOzoneVerificationDefs,
+} from '@atproto/api'
+import { Database } from '../db'
+import { CreatedAtUriKeyset, paginate } from '../db/pagination'
+import { Verification } from '../db/schema/verification'
+
+export type VerificationServiceCreator = (db: Database) => VerificationService
+
+export class VerificationService {
+  constructor(public db: Database) {}
+
+  static creator() {
+    return (db: Database) => new VerificationService(db)
+  }
+
+  async create(
+    verifications: Pick<
+      Verification,
+      'uri' | 'issuer' | 'subject' | 'handle' | 'displayName' | 'createdAt'
+    >[],
+  ) {
+    // @TODO: We should probably handle errors here?
+    await this.db.transaction(async (tx) => {
+      return Promise.allSettled(
+        verifications.map((verification) => {
+          return tx.db
+            .insertInto('verification')
+            .values(verification)
+            .onConflict((oc) => oc.doNothing())
+            .execute()
+        }),
+      )
+    })
+  }
+
+  async markRevoked({
+    uris,
+    revokedBy,
+    revokedAt,
+    revokeReason,
+  }: {
+    uris: string[]
+    revokedBy?: string
+    revokedAt?: string
+    revokeReason?: string
+  }) {
+    const now = new Date().toISOString()
+    return Promise.allSettled(
+      uris.map(async (uri) => {
+        const atUri = new AtUri(uri)
+        const res = await this.db.db
+          .updateTable('verification')
+          .set({
+            revokeReason,
+            updatedAt: now,
+            revokedAt: revokedAt || now,
+            // Allow setting revokedBy to a moderator/verifier DID and if it isn't set, default to the author of the verification record
+            revokedBy: revokedBy || atUri.host,
+          })
+          .where('uri', '=', uri)
+          .where('revokedAt', 'is', null)
+          .execute()
+
+        return res
+      }),
+    )
+  }
+
+  async list({
+    sortDirection,
+    cursor,
+    createdAfter,
+    createdBefore,
+    issuers = [],
+    subjects = [],
+    isRevoked,
+    limit = 100,
+  }: {
+    sortDirection?: 'asc' | 'desc'
+    cursor?: string
+    createdAfter?: string
+    createdBefore?: string
+    issuers?: string[]
+    subjects?: string[]
+    isRevoked?: boolean
+    limit?: number
+  }) {
+    const { ref } = this.db.db.dynamic
+
+    let qb = this.db.db.selectFrom('verification').selectAll()
+
+    if (issuers.length) {
+      qb = qb.where('issuer', 'in', issuers)
+    }
+
+    if (isRevoked !== undefined) {
+      qb = qb.where('revokedAt', isRevoked ? 'is' : 'is not', null)
+    }
+
+    if (subjects.length) {
+      qb = qb.where('subject', 'in', subjects)
+    }
+
+    if (createdAfter) {
+      qb = qb.where('createdAt', '>', createdAfter)
+    }
+
+    if (createdBefore) {
+      qb = qb.where('createdAt', '>', createdBefore)
+    }
```

Comment:
There's a fix in here for `createdBefore`.  Also making it inclusive, just to be consistent with `getEvents()`.
```suggestion
    if (createdAfter) {
      qb = qb.where('createdAt', '>=', createdAfter)
    }

    if (createdBefore) {
      qb = qb.where('createdAt', '<=', createdBefore)
    }
```